src/arrayMethodSort.js

```diff
@@ -4,8 +4,28 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compareFn =
+      compareFunction ||
+      function (aValue, bValue) {
+        const a = String(aValue);
+        const b = String(bValue);
+
+        return a < b ? -1 : a > b ? 1 : 0;
+      };
+
+    for (let i = 0; i < this.length - 1; i++) {
```

Comment:
Consider adding a check to see if the array is already sorted. This can be done by setting a flag when a swap is made, and if no swaps are made in an iteration, the array can be considered sorted and the function can return early. This would optimize the best-case time complexity to O(n) when the array is already sorted.